### PR TITLE
Reviewed all docstrings to improve readability and correctness

### PR DESCRIPTION
**Pull request by lmmx**
Subject: While reading the source code for Sphinx I began to edit the docstrings and thought I may as well review the entire library's docstrings. Also fixed a typo in one log message (deprecation warning).

No breaking changes, one single warning message with a typo was changed (in a deprecation message)

  - No breaking changes or critical/severe bugfixes so branch: 4.x

### Feature or Bugfix
- Refactoring

### Purpose
- Improved readability of code (many minor improvements) and ease of understanding what modules are doing

### Detail
- Improved wording in docstrings (grammar, spelling, better explanations for certain unclear terms)
### Relates
- No previous issue, expect this to be an uncontroversial PR

### CODE REVIEW

File: sphinx/parsers.py [hunk 1]
```diff
@@ -52,7 +52,8 @@ class Parser(docutils.parsers.Parser):
     """
 
     def set_application(self, app: "Sphinx") -> None:
-        """set_application will be called from Sphinx to set app and other instance variables
+        """
+        set_application will be called from Sphinx to set app and other instance variables
```
Comment (tk0miya) [inline sphinx/parsers.py]: Why folding? In my understanding, the first line of the docstring can follow just after quotation marks.

File: sphinx/registry.py [hunk 2]
```diff
@@ -467,7 +467,9 @@ def get_envversion(self, app: "Sphinx") -> Dict[str, str]:
 
 
 def merge_source_suffix(app: "Sphinx", config: Config) -> None:
-    """Merge source_suffix which specified by user and added by extensions."""
+    """
+    Merge any user-specified source_suffix with any added by extensions.
+    """
```
Comment (tk0miya) [inline sphinx/registry.py]: This is also. One-liner is allowed in PEP-8 (see https://www.python.org/dev/peps/pep-0008/#documentation-strings)

**Review by tk0miya — APPROVED** [review]
LGTM with nits.

**Comment by tk0miya** [comment]
Could you merge the HEAD of 4.x to this PR, please? Then CI errors will be resolved.

File: sphinx/parsers.py (hunk 1 above)
Comment (lmmx) [reply sphinx/parsers.py]: Ah I think I may have been trying to reduce line length (from 90 to 87 characters)? Happy for you (or me) to change this back if you prefer

File: sphinx/registry.py (hunk 2 above)
Comment (lmmx) [reply sphinx/registry.py]: Again I think this was my attempt to reduce line length (was not sure if there was a limit). Will revert both of these changes in a new commit (hope this is the correct way to resolve).

File: sphinx/parsers.py (hunk 1 above)
Comment (tk0miya) [reply sphinx/parsers.py]: Thank you for your explanation. Now our line length limitation is 95. So reverting it is better. thank you for accepting.
https://github.com/sphinx-doc/sphinx/blob/f9941b9402821de2d60c1e94f9ac5cf0379640d0/setup.cfg#L29-L30

**Comment by tk0miya** [comment]
Oops... flake8 still warns. Could you check this please? Then, I'll merge this.

```
./sphinx/application.py:611:96: E501 line too long (96 > 95 characters)
./sphinx/config.py:218:96: E501 line too long (96 > 95 characters)
./sphinx/testing/fixtures.py:255:1: W293 blank line contains whitespace
```

**Comment by lmmx** [comment]
Very sorry for that @tk0miya and thank you for your patience. I was in a rush to send this PR in before going on holiday this week and fixed these as soon as I got back :-) I hadn't looked at the contributor's guide when making these changes, I'll make sure to use the proper tox workflows when contributing in future. Hope this finishes the PR!

**Comment by tk0miya** [comment]
Merged. Thank you for your work! It's very helpful for me because I'm not a native speaker.

**Pull request merged**